redis_hashring/__init__.py

```diff
@@ -5,7 +5,6 @@
 import random
 import operator
 import os
-import select
```

Comment:
I think the original idea behind this is that you could use redis-hashring independently, either without gevent, or with non-monkey-patched gevent. Since this is an independent module, it's best not to make any assumptions on how the main application is set up.

By default it would use the system's select (which may or may not be gevent depending on whether you monkey-patched), but if you use `gevent_start()` it would always be gevent's `select` regardless of whether you monkey patched.

With this change you would break usage for users that prefer not to monkey-patch and use redis-hashring with gevent.

Reply:
Let me try to explain how I understand this to make sure I'm understanding it correctly.

**TL;DR:** The only change that belong in this PR is the `decode` change and the rest has to be reverted, right?

You're saying people are supposed to be allowed to use this project with `gevent` but without monkey-patching. This means if we stop using `listen` to use `get_message`, we will use the `sock.settimeout` under the hood and block the thread for that amount of time when waiting for the socket to have anything. `gevent` won't be aware of that blocking operation, since it didn't patch `socket`, and therefore the thread won't yield control.

So this means I need to revert any changes related to `get_message`, go back to using `listen` and `select.select`, and the only change that actually belongs in this PR is just the decode one. Right? This way, even if someone doesn't make `gevent` patch `select`, we will still be able to use `gevent.select` when polling our Redis pubsub subscription when running `poll` as a product of `gevent_start`.

Reply:
If that sounds right, then I undid the `get_message` change in https://github.com/closeio/redis-hashring/pull/3/commits/a0e7ab8a21a9b073fe42a82a309d5a87d742989a but still implemented draining the subscription channel before updating the node in https://github.com/closeio/redis-hashring/pull/3/commits/ccf380ca89e6984ada47b2e22eb8e0d9452b2c76. Let me know if that's not what you meant.

Reply:
Hmm. Maybe for readability reasons it's fine as long as the README mentions that the functions need to be patched when using it in gevent mode _before_ importing redis-hashring. Sorry about the back and forth, after thinking about it one more time I don't think we need to support the niche case where you don't monkey patch, but we need to be aware of it and point it out. Also, in the raw select code the subscribe message would trigger an update, which is unnecessary.

It's not a strong opinion, if we leave it with `self._select` that's fine as well. In this case though I would add a comment saying why we're not using `get_message()` so that we don't get confused in the future.

Reply:
How does it look now?